core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -81,4 +81,8 @@ private CatalogProperties() {
   public static final String APP_ID = "app-id";
   public static final String USER = "user";
 
+  public static final String IO_METRICS_IMPL = "io.metrics-impl";
+  public static final String IO_METRICS_IMPL_DEFAULT = "org.apache.iceberg.hadoop.HadoopMetricsContext";
+  public static final String IO_METRICS_NAMESPACE = "io.metrics.namespace";
```

Comment:
I find this name confusing because I don't think of a metrics namespace. This is specific to Hadoop, right?

Reply:
I think the best solution here is to not define a public property for "namespace". I think the IO classes should just pass a "scheme" option into the metrics class. No need to define a contract or to try to make this generic.